src/bors/handlers/workflow.rs

```diff
@@ -61,12 +62,30 @@ pub(super) async fn handle_workflow_started(
 pub(super) async fn handle_workflow_completed(
     repo: Arc<RepositoryState>,
     db: Arc<PgDbClient>,
-    payload: WorkflowCompleted,
+    mut payload: WorkflowCompleted,
 ) -> anyhow::Result<()> {
     if !is_bors_observed_branch(&payload.branch) {
         return Ok(());
     }
 
+    if let Some(running_time) = payload.running_time {
+        let running_time_as_duration =
+            chrono::Duration::to_std(&running_time).unwrap_or(Duration::from_secs(0));
+        if running_time_as_duration < repo.config.load().min_ci_time {
+            payload.status = WorkflowStatus::Failure;
+            tracing::warn!(
+                "Workflow running time is less than the minimum CI duration: {:?} < {:?}",
+                payload.running_time,
+                repo.config.load().min_ci_time
+            );
+            return Err(anyhow::anyhow!(
```

Comment:
We should not bail out here, but rather continue with the execution of the function. Currently, we won't be able to distinguish a normal failure with a "too short duration" failure, but I suppose that we can do that in another PR, e.g. by generalizing `WorkflowStatus` to include this information there.

Reply:
But the requirement is that if it is failing to give us some minimum time build should not go through rather it should fail. Can you give more information about what should happen. Should it not return a Error.

Reply:
It does not need to return an error. The fact that a workflow has failed is not an error, that's normal operation :) We store that information into the DB and continue executing.